includes/multi-currency/Compatibility/WooCommerceProductAddOns.php

```diff
@@ -110,37 +111,76 @@ public function product_addons_params( array $params ): array {
 	 */
 	public function get_item_data( $addon_data, $addon, $cart_item ): array {
 		$price = isset( $cart_item['addons_price_before_calc'] ) ? $cart_item['addons_price_before_calc'] : $addon['price'];
-		$name  = $addon['name'];
-
-		if ( 0.0 === $addon['price'] ) {
-			$name .= '';
-		} elseif ( 'percentage_based' === $addon['price_type'] && 0.0 === $price ) {
-			$name .= '';
-		} elseif ( 'custom_price' === $addon['field_type'] ) {
-			$name .= ' (' . wc_price( $addon['price'] ) . ')';
-		} elseif ( 'percentage_based' !== $addon['price_type'] && $addon['price'] && apply_filters( 'woocommerce_addons_add_price_to_name', '__return_true' ) ) {
-			// Get our converted and tax adjusted price to put in the add on name.
-			$price = $this->multi_currency->get_price( $addon['price'], 'product' );
-			if ( 'input_multiplier' === $addon['field_type'] ) {
-				// Quantity/multiplier add on needs to be split, calculated, then multiplied by input value.
-				$price = $this->multi_currency->get_price( $addon['price'] / $addon['value'], 'product' ) * $addon['value'];
+		$value = $addon['value'];
+
+		/*
+		 * 'woocommerce_addons_add_cart_price_to_value'
+		 *
+		 * Use this filter to display the price next to each selected add-on option.
+		 * By default, add-on prices show up only next to flat fee add-ons.
+		 *
+		 * @param boolean
+		 */
+		$add_price_to_value = apply_filters( 'woocommerce_addons_add_cart_price_to_value', false, $cart_item );
+
+		if ( 0.0 === (float) $addon['price'] ) {
+			$value .= '';
+		} elseif ( 'percentage_based' === $addon['price_type'] && 0.0 === (float) $price ) {
+			$value .= '';
+		} elseif ( 'custom_price' === $addon['field_type'] && $addon['price'] ) {
+			if ( class_exists( '\WC_Product_Addons_Helper' ) ) {
+				// phpcs:ignore
+				/**
+				 * @psalm-suppress UndefinedClass
+				 */
```

Comment:
Nit: Can we update this to the following?
```suggestion
			if ( class_exists( 'WC_Product_Addons_Helper' ) ) {
```

Reply:
Good idea! Resolved.